contracts/JBETHPaymentTerminalStore.sol

```diff
@@ -476,8 +476,16 @@ contract JBETHPaymentTerminalStore {
       ? _amount
       : PRBMathUD60x18.div(_amount, prices.priceFor(_currency, JBCurrencies.ETH));
 
-    // The amount being withdrawn must be available.
-    if (withdrawnAmount > balanceOf[_projectId]) {
+    // Get the current funding target
+    uint256 distributionLimit =
+      directory.controllerOf(_projectId).distributionLimitOf(
+        _projectId,
+        fundingCycle.configuration,
+        terminal
+      );
+
+    // The amount being withdrawn must be available in the overflow.
+    if (withdrawnAmount > balanceOf[_projectId] + usedDistributionLimitOf[_projectId][fundingCycle.number] - distributionLimit) {
```

Comment:
nice catch!

I'm wondering if we should also account for the overflow that can be spread out across multiple terminals like it's done here: https://github.com/jbx-protocol/juice-contracts-v2/blob/e0722447d79de0af79017e6057065bcf9f845ee9/contracts/JBETHPaymentTerminalStore.sol#L677-L681

@mejango thoughts?